core-bundle/src/Image/Studio/Figure.php

```diff
@@ -0,0 +1,378 @@
+    {
+        $this->resolveIfClosure($this->metaData);
+
+        return $this->metaData instanceof MetaData;
+    }
+
+    /**
+     * Return the main resource's meta data.
+     */
+    public function getMetaData(): MetaData
+    {
+        if (!$this->hasMetaData()) {
+            throw new \LogicException('This result container does not include meta data.');
+        }
+
+        // Safely return as Closure will be evaluated by this point
+        return $this->metaData;
+    }
+
+    /**
+     * Return a key-value list of all link attributes (excluding `href` by default).
+     */
+    public function getLinkAttributes(bool $includeHref = false): array
+    {
+        $this->resolveIfClosure($this->linkAttributes);
+
+        if (null === $this->linkAttributes) {
+            $this->linkAttributes = [];
+        }
+
+        // Generate href attribute
+        if (!\array_key_exists('href', $this->linkAttributes)) {
+            $this->linkAttributes['href'] = (
+            function () {
+                if ($this->hasLightBox()) {
+                    return $this->getLightBox()->getLinkHref();
+                }
+
+                if ($this->hasMetaData()) {
+                    return $this->getMetaData()->getUrl();
+                }
+
+                return '';
+            }
+            )();
+        }
+
+        // Add rel attribute ("noreferrer noopener") to external links
+        if (
+            !empty($this->linkAttributes['href']) &&
+            !\array_key_exists('rel', $this->linkAttributes) && preg_match('#^https?://#', $this->linkAttributes['href'])
+        ) {
+            $this->linkAttributes['rel'] = 'noreferrer noopener';
+        }
+
+        // Add light box attributes
+        if (!\array_key_exists('data-lightbox', $this->linkAttributes) && $this->hasLightBox()) {
+            $lightBox = $this->getLightBox();
+            $this->linkAttributes['data-lightbox'] = $lightBox->getGroupIdentifier();
+        }
+
+        // Allow removing attributes by setting them to `null`
+        $linkAttributes = array_filter(
+            $this->linkAttributes,
+            static function ($attribute): bool {
+                return null !== $attribute;
+            }
+        );
+
+        // Optionally strip href attribute
+        return $includeHref ? $linkAttributes : array_diff_key($linkAttributes, ['href' => null]);
+    }
+
+    /**
+     * Return the `href` link attribute.
+     */
+    public function getLinkHref(): string
+    {
+        return $this->getLinkAttributes(true)['href'] ?? '';
+    }
+
+    /**
+     * Return a key-value list of (figure) attributes.
+     */
+    public function getOptions(): array
+    {
+        $this->resolveIfClosure($this->options);
+
+        return $this->options ?? [];
+    }
+
+    /**
+     * Compile an opinionated data set ready to be applied to a Contao template.
+     *
+     * Note: Do not use this method when using modern/Twig templates! Instead,
```

Comment:
What are "modern" templates that are not Twig templates?

Reply:
You could also create new PHP templates that are able to deal with a `Figure` objects directly. E.g. a `figure.html5` as successor of the `image.html5`.

Reply:
Clarified the comment in 4879b5bd52a8bb61c638319acaf3d0b55354ff96.